subscriber/synapse-subscriber-kafka/src/main/java/io/americanexpress/synapse/subscriber/kafka/config/BaseKafkaPropertiesConfiguration.java

```diff
@@ -0,0 +1,224 @@
+     */
+    private final Environment environment;
+
+    /**
+     * This method is used to get bootstrapServers.
+     */
+    @Override
+    public List<String> getBootstrapServers() {
+        return Collections.singletonList(environment.getRequiredProperty("kafka.bootstrap.servers"));
+    }
+
+    /**
+     * This method is used to get kafka Ssl.
+     */
+    @Override
+    public Ssl getSsl() {
+        return this.ssl;
+    }
+
+    /**
+     * This method is used to get kafka consumer.
+     */
+    @Override
+    public Consumer getConsumer() {
+        return this.consumer;
+    }
+
+    /**
+     * This method is used to build a HashMap of consumer properties.
+     */
+    @Override
+    public Map<String, Object> buildConsumerProperties() {
+        return this.getConsumer().buildProperties();
+    }
+
+
+    /**
+     * <code>BaseKafkaConsumer</code> class contains the properties related to a consumer configuration.
+     */
+    @RequiredArgsConstructor
+    public abstract static class BaseKafkaConsumer extends Consumer {
+
+        /**
+         * Environment.
+         */
+        private final Environment environment;
+
+        /**
+         * Bootstrap Server.
+         */
+        @Override
+        public List<String> getBootstrapServers() {
+            return Collections.singletonList(environment.getRequiredProperty("kafka.bootstrap.servers"));
+        }
+
+        /**
+         * Auto-commit.
+         */
+        @Override
+        public Boolean getEnableAutoCommit() {
+            return false;
+        }
+
+        /**
+         * This method is used to get de-serializer used for key of the message.
+         */
+        @Override
+        public Class<?> getKeyDeserializer() {
+            return StringDeserializer.class;
+        }
+
+        /**
+         * This method is used to get de-serializer used for value of the message.
+         */
+        @Override
+        public Class<?> getValueDeserializer() {
+            return StringDeserializer.class;
+        }
+
+        /**
+         * This method is used to get maximum number of records to be polled.
+         */
+        @Override
+        public Integer getMaxPollRecords() {
+            return Optional.ofNullable(environment.getProperty("kafka.subscriber.maximum.poll.records", Integer.class)).orElse(1);
+        }
+    }
+
+    /**
+     * <code>BaseKafkaSSL</code> class contains the properties related to a ssl configuration.
+     */
+    @RequiredArgsConstructor
+    public abstract static class BaseKafkaSsl extends Ssl {
+
+        /**
+         * Environment.
+         */
+        private final Environment environment;
+
+        /**
+         * ResourceLoader.
+         */
+        private final ResourceLoader resourceLoader;
+
+        /**
+         * This method is used to get key password.
+         */
+        @Override
+        public String getKeyPassword() {
+            return environment.getRequiredProperty("key.identification");
```

Comment:
thinking if this should be kafka.key.password?

Reply:
I have used `identification` in place of `password` as it maybe scanned as a secret in readMe documents. 
Please let me know if we can use the keyword `password`

Reply:
I think we should use password

Reply:
use the default key names
https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html

Reply:
done